app/services/finance/charging_service.rb

```diff
@@ -36,14 +36,14 @@ def charge_with_authorize_net
       payment_profile = payment_profiles.is_a?(Array) ? payment_profiles[-1] : payment_profiles # payment_profiles can be a Hash or an Array of hashes
       payment_profile_id = payment_profile['customer_payment_profile_id']
 
-      gateway.cim_gateway.create_customer_profile_transaction(
-        transaction: {
-          customer_profile_id: buyer_reference,
-          customer_payment_profile_id: payment_profile_id,
-          type: :auth_capture,
-          amount: amount.to_f
-        }
-      )
+      gateway.cim_gateway.create_customer_profile_transaction({
```

Comment:
In this case I chose to "fix" the code, and not the test, because I think this is more correct.
`create_customer_profile_transaction` is a method of Active Merchant's `AuthorizeNetCimGateway`, and in their test they use both hash and keyword arguments:
https://github.com/activemerchant/active_merchant/blob/v1.107.4/test/remote/gateways/remote_authorize_net_cim_test.rb

I think both ways work, because this is kind of an exception:
![Screenshot from 2025-01-23 15-48-03](https://github.com/user-attachments/assets/838c85eb-306c-4cbb-bd36-27bb1d981feb)

https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

But I think passing a hash here is less confusing.